api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirLocationDaoImpl.java

```diff
@@ -38,6 +41,8 @@
 @Setter(AccessLevel.PACKAGE)
 public class FhirLocationDaoImpl extends BaseFhirDao<Location> implements FhirLocationDao {
 	
+	private static final int SUPPORTED_LOCATION_HIERARCHY_SEARCH_DEPTH = 9;
```

Comment:
Basically, I use a succession of joins to go back up the ancestor tree.  A bit annoying/hacky, but from a practical matter, seems unlikely there would ever be a hierarchy of more than 9 levels.  We could make this a configurable parameter if we wanted.

Reply:
Maybe we turn this into a GP? That way an implementation can adjust it to the actual depth of their location hierarchy.

Reply:
Fair enough, I will do that (and since it's configurable, set a more normal default of like 5)